### PR TITLE
Domain Name needs to be passed to the entrypoint.sh for a cross cloud Pyrsia network

### LINKED ISSUE #1544: entrypoint.sh needs to know the domain name for external-dns
The entrypoint.sh needs to know which domain the node is being register under.  The current implementation uses the bootdns name but that does not allow for cross cloud nodes.  The helm charts already define the domain and bootdns as separate variables in the values file.  The entrypoint.sh needs these values so it can correctly register the nodes dns name.

### LINKED ISSUE #1554: Update release doc to include AWS
The release doc needs instructions on how to connect kubectl to both gke and eks.   The helm commands will be the same between the two environments except:
1. buildnode parameter will be different
2. domain parameter will be different
3. keypairs will be different

### PR DESCRIPTION
**Pull request by sbtaylor15**
## Description

Changed the helm charts, entrypoint.sh and doc to reflect the domain name being passed from the chart to the entrypoint.sh so the node can register its dns name correctly.  Also, updated the doc to use variables instead of hard coding.

Added instructions for EKS and GKE instructions to both staging and prod.

Fixes pyrsia/pyrsia#1544
Fixes #1554

## PR Checklist

- [X] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [X] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/community/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [X] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've run [pre-commit.sh](https://github.com/pyrsia/pyrsia/blob/main/pre-commit.sh) ([pre-commit.bat](https://github.com/pyrsia/pyrsia/blob/main/pre-commit.bat)) successfully.
- [x] I've run the [integrations tests](https://github.com/pyrsia/pyrsia-integration-tests#how-to-set-up-and-run-the-tests)
on the PR branch and all tests passes.

### CODE REVIEW

File: docs/SRE/eks-setup.md [hunk 1]
```diff
@@ -41,35 +83,35 @@
...
+   }'
+   ```
+
+12. Deploy Pyrsia via Helm
 
-11. Deploy Pyrsia via Helm
-      - `helm repo update pyrsia-nightly`
-      - `helm upgrade node1 --install -n pyrsia-node pyrsia-nightly/pyrsia-node --set k8s_provider=eks --set "dnsname=nightly.pyrsia-aws.link" --set bootdns=boot.nightly.pyrsia-aws.link --set keys.p2p=$(cat ed25519.ser | base64) --set keys.blockchain=$(cat ed25519.ser | base64)  --version "0.2.4+2856`
+   - `helm repo update pyrsia-nightly`
+   - `helm upgrade node1 --install -n pyrsia-node pyrsia-staging/pyrsia-node --set k8s_provider=eks --set "domain=${PYRSIA_DOMAIN}" --set bootdns=${PYRSIA_BOOTDNS} --set keys.p2p=$(cat ed25519.ser | base64) --set keys.blockchain=$(cat ed25519.ser | base64)  --version "0.2.4+2856`
```
Comment (dasmanas) [inline docs/SRE/eks-setup.md:115]: Is it possible to create variable for version as well?

File: docs/SRE/gke-setup.md [hunk 2]
```diff
@@ -49,8 +63,8 @@
    cat id_ed25519.pri id_ed25519.pub > ed25519.ser
    ```
 
-10. Deploy Pyrsia via Helm
+11. Deploy Pyrsia via Helm
       - `helm repo update pyrsia-nightly`
-      - `helm upgrade node1 --install -n pyrsia-node pyrsia-nightly/pyrsia-node --set k8s_provider=gke --set "dnsname=staging.pyrsia.link" --set bootdns=boot.staging.pyrsia.link --set keys.p2p=$(cat ed25519.ser | base64) --set keys.blockchain=$(cat ed25519.ser | base64) --version "0.2.4+2856`
+      - `helm upgrade node1 --install -n pyrsia-node pyrsia-nightly/pyrsia-node --set k8s_provider=gke --set "domain=${PYRSIA_DOMAIN}" --set bootdns=${PYRSIA_BOOTDNS} --set keys.p2p=$(cat ed25519.ser | base64) --set keys.blockchain=$(cat ed25519.ser | base64) --version "0.2.4+2856`
```
Comment (dasmanas) [inline docs/SRE/gke-setup.md:68]: Is it possible to create variable for version as well?

File: docs/SRE/kubernetes-helm.md [hunk 3]
```diff
@@ -87,7 +87,7 @@ Details about the Chart Values and Installation steps are documented in the char
         docker build --tag mydockerhubid/pyrsia:1.0
         docker push mydockerhubid/pyrsia:1.0
 
-        helm upgrade --install node1 -n pyrsia-node pyrsia-nightly/pyrsia-node --set "k8s_provider=gke" --set "dnsname=staging.pyrsia.link" --set "bootdns=boot.staging.pyrsia.link"  --set "replicaCount=1" --set "buildnode=http://35.193.148.20:8080" --set image.repository=mydockerhubid --set image.tag=1.0 --set keys.p2p=$(cat ed25519.ser | base64) --set keys.blockchain=$(cat ed25519.ser | base64) --version "0.2.1+2562"
+        helm upgrade --install node1 -n pyrsia-node pyrsia-nightly/pyrsia-node --set "k8s_provider=gke" --set "domain=staging.pyrsia.link" --set "bootdns=boot.staging.pyrsia.link"  --set "replicaCount=1" --set "buildnode=http://35.193.148.20:8080" --set image.repository=mydockerhubid --set image.tag=1.0 --set keys.p2p=$(cat ed25519.ser | base64) --set keys.blockchain=$(cat ed25519.ser | base64) --version "0.2.1+2562"
```
Comment (dasmanas) [inline docs/SRE/kubernetes-helm.md:90]: If build is setup before hand, can we replace the hardcoded ip with any node selector query or something like that. Just a suggestion.

**Review by dasmanas — DISMISSED** [review]
Lgtm. Approved with some comments.

File: docs/SRE/eks-setup.md (hunk 1 above)
Comment (sbtaylor15) [reply docs/SRE/eks-setup.md]: added CHART_VERSION

File: docs/SRE/kubernetes-helm.md (hunk 3 above)
Comment (sbtaylor15) [reply docs/SRE/kubernetes-helm.md:90]: added CHART_VERSION

Comment (sbtaylor15) [reply docs/SRE/kubernetes-helm.md:90]: FYI, there is no way to derive the build node at this time

File: docs/SRE/gke-setup.md (hunk 2 above)
Comment (sbtaylor15) [reply docs/SRE/gke-setup.md]: add CHART_VERSION

**Review by efrisch — APPROVED** [review]
I like the environment variables in the markdown, thanks!
lgtm. 👍

File: docs/SRE/kubernetes-helm.md [hunk 4]
```diff
@@ -64,36 +64,145 @@ Details about the Chart Values and Installation steps are documented in the char
...
+
 2. Fetch the latest charts
 `helm repo update`
+
 3. Set you cluster connection
 `kubectl config use-context <context name>`
-4. Deployment
 
-    > Note: Change the --version of the chart to reflect the image tag you want to deploy.  The image tag and chart version are kept in sync.
+4. Obtain the Key Pairs from Last Pass
```
Comment (fishseabowl) [inline docs/SRE/kubernetes-helm.md:74]: FYI, [recently lastpass suffered a data breach and customer backups were compromise](https://thehackernews.com/2023/01/lastpass-parent-company-goto-suffers.html)

**Review by fishseabowl — APPROVED** [review]
lgtm

**Pull request merged**